addons/point_of_sale/models/pos_session.py

```diff
@@ -101,6 +101,265 @@ class PosSession(models.Model):
+                'fields': ['id', 'product_id', 'combo_price', 'combo_id']
+            },
+            'product.packaging': {
+                'domain': lambda data: AND([[('barcode', 'not in', ['', False])], [('product_id', 'in', [x['id'] for x in data['product.product']])] if data else []]),
+                'fields': ['id', 'name', 'barcode', 'product_id', 'qty'],
+            },
+            'res.users': {
+                'domain': [('id', '=', self.env.user.id)],
+                'fields': ['id', 'name', 'groups_id', 'partner_id'],
+            },
+            'res.partner': {
+                'domain': [('id', 'in', config_id.get_limited_partners_loading())],
+                'fields': [
+                    'id',
+                    'name', 'street', 'city', 'state_id', 'country_id', 'vat', 'lang', 'phone', 'zip', 'mobile', 'email',
+                    'barcode', 'write_date', 'property_account_position_id', 'property_product_pricelist', 'parent_name'
+                ]
+            },
+            'res.company': {
+                'domain': [('id', '=', config_id.company_id.id)],
+                'fields': [
+                    'id',
+                    'currency_id', 'email', 'website', 'company_registry', 'vat', 'name', 'phone', 'partner_id',
+                    'country_id', 'state_id', 'tax_calculation_rounding_method', 'nomenclature_id', 'point_of_sale_use_ticket_qr_code',
+                    'point_of_sale_ticket_unique_code', 'street', 'city', 'zip',
+                ],
+            },
+            'decimal.precision': {
+                'domain': [],
+                'fields': ['id', 'name', 'digits'],
+            },
+            'uom.uom': {
+                'domain': [],
+                'fields': ['id', 'name', 'category_id', 'factor_inv', 'factor', 'is_pos_groupable', 'uom_type', 'rounding'],
+            },
+            'uom.category': {
+                'domain': lambda data: [('uom_ids', 'in', [uom['category_id'] for uom in data['uom.uom']])],
+                'fields': ['id', 'name', 'uom_ids'],
+            },
+            'res.country.state': {
+                'domain': [],
+                'fields': ['id', 'name', 'code', 'country_id'],
+            },
+            'res.country': {
+                'domain': [],
+                'fields': ['id', 'name', 'code'],
+            },
+            'res.lang': {
+                'domain': [],
+                'fields': ['id', 'name', 'code'],
+            },
+            'product.pricelist' : {
+                'domain': [('id', 'in', config_id.available_pricelist_ids.ids)] if config_id.use_pricelist else [('id', '=', config_id.pricelist_id.id)],
+                'fields': ['id', 'name', 'display_name', 'discount_policy', 'item_ids']
+            },
+            'product.pricelist.item' : {
+                'domain': lambda data: [('pricelist_id', 'in', [pricelist['id'] for pricelist in data['product.pricelist']])],
```

Comment:
Load only `pricelist.item`s that are linked in the loaded products just like what is being done in: https://github.com/odoo/odoo/blob/b02faa688234f8e7ccaac211678d3a23e5f3764c/addons/point_of_sale/models/pos_session.py#L2203C23-L2203C23